tags.go

```diff
@@ -200,3 +206,66 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	// Get underlying resource type for type checking
+	switch resource := resourceType.(type) {
+	case *Domain:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *InfrastructureResource:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Repository:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Service:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *System:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Team:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *User:
+		tagConnection, err = resource.GetTags(client, nil)
+	default:
+		return assignedTags, fmt.Errorf("not a taggable resource type: %s" + string(resourceType.ResourceType()))
+	}
+	if err != nil {
+		return assignedTags, err
```

Comment:
[non-blocking] I suggest wrapping these errors since there are different API client calls that cause them